pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -241,6 +252,14 @@ func (a *Adapter) Update(ctx context.Context, updateOp *directbase.UpdateOperati
 	}
 	wp.Name = a.id.FullyQualifiedName()
 	wp.Etag = a.actual.Etag
+	// the peered_network has a different HTTP request and response format.
+	// The HTTP request uses ProjectID, in the form of /projects/<projectID>/global/networks/<network>
```

Comment:
Can we avoid this by passing in the project number?

Reply:
we want to use the compareProto.

Reply:
No, I mean that peeredNetworkRef is a ComputeNetworkRef https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/86da25427f1a05f46a9cf7a8078d8ad61409140f/apis/cloudbuild/v1beta1/workerpool_types.go#L79

So what I'm suggesting is that we use something like LinkWithProjectNumber() instead of String() here: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/86da25427f1a05f46a9cf7a8078d8ad61409140f/pkg/controller/direct/cloudbuild/workerpool_controller.go#L130

The downside of doing this is that we also want to work towards resolving references using a generic visitor, and that's going to be harder if sometimes we want link-with-project-id and sometimes link-with-project-number, but I think we want to have correctness first and then make the resolving code elegant.

Reply:
This project ID/Num specialty seems to be less relevant to `ComputeNetwork` but more a CBWP problem. I looked at the ComputeNetwork object, it does not use ProjectNumber , see ` gcloud compute networks describe default`.  We can add a `LinkWithProjectID` method, in case other resource does the same thing. But I want to keep it as a special handling inside CBWP than adding the solution in ComputeNetwork. Because the computeNetwork has its own issues that we need to fix.